File: Swim/Swim/Infrastructure/DatabaseConnection.cs
﻿using System.Data.SqlClient;

namespace Swim.Infrastructure
{
    public class DatabaseConnection
    {
        private readonly string connectionString = @"Server=LAPTOP-16N38ULV\SQLEXPRESS;Database=MPP
            ;Integrated Security=true;TrustServerCertificate=true";

        private SqlConnection? connection;

        DatabaseConnection() { }
        private static readonly object _lock = new();  
        private static DatabaseConnection? instance = null;
        public static DatabaseConnection Instance
        {
            get
            {
                lock (_lock)
                    {
                        if (instance == null)
                        {
                            instance = new DatabaseConnection();
                        }
                        return instance;
                    }
            }
        }


        public void Open()
        {
            if (connection == null)
                return;

            try
            {
                connection = new SqlConnection(connectionString);
                connection.Open();

            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        public SqlConnection? Get()
        {
            return connection;
        }
    }
}

File: Java/src/main/java/swim/Data/Utils/DatabaseConnection.java
package swim.Data.Utils;

import java.sql.Connection;
import java.sql.DriverManager;

public class DatabaseConnection
{
    private static Connection Instance;
    private static final String connectionString = "jdbc:sqlserver://localhost;database=MPP;encrypt=true;trustServerCertificate=true;";
    private static final String username = "admin";
    private static final String password = "admin";
    private Connection connection;

    private static DatabaseConnection instance = null;

    DatabaseConnection() {
        Open();
    }

    public static synchronized DatabaseConnection getInstance()
    {
        if (instance == null)
            instance = new DatabaseConnection();

        return instance;
    }

    private void Open()
    {
        try
        {
            connection = DriverManager.getConnection(connectionString, username, password);
        }
        catch (Exception ex) {
            throw new RuntimeException(ex);
        }
    }

    public Connection Get()
    {
        return connection;
    }
}
